src/derive/field.rs

```diff
@@ -686,3 +686,38 @@ macro_rules! field_bits {
         }
     };
 }
+
+/// A macro to help define serialization and deserialization for prime field implementations
+/// that use 32-byte representations. This assumes the concerned type implements PrimeField
+/// (for from_repr, to_repr).
+#[macro_export]
+macro_rules! serialize_deserialize_32_byte_primefield {
+    ($type:ty) => {
+        impl ::serde::Serialize for $type {
+            fn serialize<S: ::serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+                let bytes = &self.to_repr();
+                if serializer.is_human_readable() {
+                    hex::serde::serialize(bytes, serializer)
+                } else {
+                    bytes.serialize(serializer)
+                }
```

Comment:
I'm afraid this `if` will even make performance worse when it comes to serialization.

On another hand, this never has anything to do with raw serialization (which is usually the used for performance). 
Do I understand correctly that we're using `serde` then as "debugging serialization"??

Reply:
Serializers usually just implement `is_human_readable` as a constant, so this `if` is as cheap as can be. Examples:
- [the default, inlined and often not overridden](https://github.com/serde-rs/serde/blob/ddc1ee564b33aa584e5a66817aafb27c3265b212/serde/src/ser/mod.rs#L1435)
- [bincode](https://github.com/bincode-org/bincode/blob/2f08107156e378b7588323b622adf4695db33d08/src/features/serde/ser.rs#L276)
- [cborium](https://github.com/enarx/ciborium/blob/fa7421546b2ffeda181823fb60e57fd81ffd227c/ciborium/src/tag.rs#L423-L426)

The semantics of `is_human_readable`, [as the doc indicates](https://docs.rs/serde/latest/serde/ser/trait.Serializer.html#method.is_human_readable), is not debugging. It's simply articulating the difference between formats that are meant to be human-readable (JSON, YAML), and those that are not (bincode, CBOR, RLP, ...).

Reply:
> Serializers usually just implement is_human_readable as a constant, so this if is as cheap as can be. 

Fair point.